### PR TITLE
Feature/iid  implement 'iid' sampling of feature vectors from RDDs

### PR DESCRIPTION
**Pull request by erikerlandson**

### CODE REVIEW

**Comment by erikerlandson** [comment]
Note, this is written on top of the outstanding PR #10 for feature extraction combinators

**Comment by erikerlandson** [comment]
Rebased off of the latest `develop` branch with Extractor PR

**Comment by erikerlandson** [comment]
Possibly worth considering an alternative to 'iid', because synthetic data is independently distributed, but won't in general be identically distributed.   Not sure what a good replacement name would be.

**Comment by erikerlandson** [comment]
Maybe `mdsFeatureSeqRDD`, mds is for "Marginal Distribution Sampling"

File: src/main/scala/com/redhat/et/silex/sample/iid/rdd/IIDFeatureSamplingMethodsRDD.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    require(n == 0 || (iSS > 0 && oSS > 0))
+
+    logInfo("Computing data sampling ratio")
+    val N = data.count
+    require(n == 0 || N > 0L)
+
+    val M = data.first.length
+    val fraction = if (N > 0L) math.min(1.0, iSS.toDouble / N.toDouble) else 0.0
+
+    var remaining = n
```
Comment (willb) [inline src/main/scala/com/redhat/et/silex/sample/iid/rdd/IIDFeatureSamplingMethodsRDD.scala:51]: I think we can clean up this code a bit but am not going to hold up the PR for it.

**Pull request closed without merging**